src/GraphQL/Reflection/ReflectionHelper.cs

```diff
@@ -98,11 +98,13 @@ public static PropertyInfo PropertyForField(this Type type, string field)
             }
 
             if (parameters.Length > index
-                && context.UserContext != null
-                && context.UserContext?.GetType() == parameters[index].ParameterType)
```

Comment:
Do you mind to move `parameters[index].ParameterType.IsInstanceOfType(context.UserContext)` condition right into this line of code to minimize changes?

Reply:
sorry i dont really groc your comment. can you make the change?

Reply:
done

Reply:
thanks